packages/itwinui-react/src/core/Table/ColumnHeader.tsx

```diff
@@ -47,16 +45,15 @@ export const ColumnHeader = <
   const {
     columnRefs,
```

Comment:
Can `columnRefs` be removed? It doesn't feel quite right for a `ColumnHeader` to mutate its parent component's variables.

Reply:
I have removed this prop.

Reply:
You have removed the whole functionality, not just the prop. What I was asking for is to move the `columnRefs`-related code from `ColumnHeader` into `Table`. 

You can pass a `ref` from the outside.

```js
ref={(el) => {
  if (el) {
    columnRefs.current[column.id] = el;
  }
}}
```

And the refs will need to be merged inside `ColumnHeader`.

Reply:
I have the same concern regarding generic type `T` when making this a `forwardRef` component to pass in `ref` from its parent.

Reply:
I don't think we even need to use a generic in `ColumnHeader`. We can get rid of it because it's an internal component. In the places where `T` is used inside the component, we can use `Record<string, unknown>`.

Reply:
I have replaced `T` with `Record<string, unknown>`.

Reply:
I noticed two things in `ColumnHeader.tsx`:
1. The ref callback is no longer memoized. This undoes the performance improvement from https://github.com/iTwin/iTwinUI/pull/2178.
2. (nit) The forwarded ref is vaguely named `ref`, when it should instead be `forwardedRef`.

The code should look something like this:

```jsx
ref={useMergedRefs(
  React.useCallback((el: HTMLDivElement) => {
    if (el) {
      column.resizeWidth = el.getBoundingClientRect().width;
    }
  }, [column]),
  forwardedRef
)}
```

Reply:
I have made this change.